processor/src/main/kotlin/permissions/dispatcher/processor/util/Extensions.kt

```diff
@@ -99,5 +115,19 @@ fun FileSpec.Builder.addTypes(types: List<TypeSpec>): FileSpec.Builder {
  * To avoid KotlinPoet bug that returns java.lang.String when type name is kotlin.String.
  * This method should be removed after addressing on KotlinPoet side.
  */
-fun TypeName.checkStringType() =
-        if (this.toString() == "java.lang.String") ClassName("kotlin", "String") else this
+fun TypeName.mapPrimitivesToKotlinTypes() =
+        when (this.toString()) {
+            "java.lang.String" -> ClassName("kotlin", "String")
```

Comment:
I think primitive types except `String` are well-converted in [KotlinPoet](https://github.com/square/kotlinpoet/blob/2bca828f131c7f2274246e91639164abfb7a9fb3/src/main/java/com/squareup/kotlinpoet/TypeName.kt#L124). Are there any necessity to take care?

Reply:
When it came to nullable primitives, I remember that the generated code favored `@Nullable java.lang.Integer` over `kotlin.Int?`, which was weird. Maybe it's an issue with KotlinPoet? I'll try to reproduce and open a ticket

Reply:
Oh really? I thought that was only a problem with `java.lang.String`? 👀 
I haven't tested in all primitive types so not sure but if the convert is weird it's ok to keep for sure 🙇 
ref: https://github.com/square/kotlinpoet/issues/236

Reply:
Sorry for the late, any update for this? 👁

Reply:
Not yet, sorry! I'll try to find a time slot to verify this behavior

Reply:
No problem 🙇